### PR TITLE
Improve codebase for Windows regular apps and appX, better document Microsoft Store details + improve Linux .desktop filename

### PR DESCRIPTION
**Pull request by Oxalin**
- Target platforms this affects: Linux, Windows
- What problem does this solve? Simplify code maintenance for Windows and make .desktop file more coherent with filenames generated by electron.
- Could it break any existing functionality for users? No. However, for Linux, it could change the .desktop filename and necessitate the app to re-enable auto-launch.

Codebases from both Microsoft Store appX and regular apps are unified under a single common codebase.

Reintroduce some details about Microsoft Store appX and how to identify the path value to use with auto-launch.

### CODE REVIEW

**Comment by Oxalin** [comment]
I added some tests to make sure the path is correctly escaped under Linux and that we honor the appName (name).

**Comment by Oxalin** [comment]
I think we could bump the package version after that.

**Comment by Oxalin** [comment]
@4ver Would you be able to review this PR please?

File: src/AutoLaunchLinux.coffee [hunk 1]
```diff
@@ -25,18 +25,18 @@ module.exports =
...
     # appName - {String}
     # Returns a Promise
-    disable: (appName) -> fileBasedUtilities.removeFile @getFilePath appName
+    disable: (appName) -> fileBasedUtilities.removeFile @getFilePath appName.toLowerCase()
 
 
     # appName - {String}
     # Returns a Promise which resolves to a {Boolean}
-    isEnabled: (appName) -> fileBasedUtilities.isEnabled @getFilePath appName
+    isEnabled: (appName) -> fileBasedUtilities.isEnabled @getFilePath appName.toLowerCase()
```
Comment (4ver) [inline src/AutoLaunchLinux.coffee:39]: Why are you lowercasing the appName?

Would be good to make a test with different app name cases to test that this still works

**Review by 4ver — CHANGES_REQUESTED** [review]

File: src/AutoLaunchLinux.coffee (hunk 1 above)
Comment (4ver) [reply src/AutoLaunchLinux.coffee:39]: Could you pull from master so tests will run for your commits?

Comment (Oxalin) [reply src/AutoLaunchLinux.coffee:39]: > Why are you lowercasing the appName?
> 
> Would be good to make a test with different app name cases to test that this still works

Good question. I've been going on and off about it. There is nothing requiering the filename to be in lowercase. And we can also go in favor with the fact that we should honor how the appName sent to us. On the other hand, most apps go with lowercase filename for Desktop file. This wouldn't be a problem if we were to support a parameter for the appName and one for the Desktop file's name, but we are not offering this option at this point.

For testcases, I think we should at least add a test where the appName could have a space in it, which should be a supported scenario.

Comment (Oxalin) [reply src/AutoLaunchLinux.coffee:39]: > Could you pull from master so tests will run for your commits?

You want me to pull in my PR or you want me to pull it out? Testcases were succeeding in my fork.

Comment (Oxalin) [reply src/AutoLaunchLinux.coffee]: I chose to respect case by: https://github.com/Teamwork/node-auto-launch/pull/120/commits/2483d2b53388d10e98d4633daee09e4282755761

Comment (Oxalin) [reply src/AutoLaunchLinux.coffee]: @4ver I'll need your review to be able to pull in the series. Your comments are welcomed.

Comment (4ver) [reply src/AutoLaunchLinux.coffee]: > I've been going on and off about it. There is nothing requiering the filename to be in lowercase. And we can also go in favor with the fact that we should honor the appName sent to us. On the other hand, most apps go with lowercase filename for Desktop file. 

Not sure i get you. On linux, the filesystem is case-sensitive so there could technically be two files with the same name but with different cases. Probably best to go with what's been passed as appName?

> You want me to pull in my PR or you want me to pull it out? Testcases were succeeding in my fork.

Ah sorry! I haven't set it up to run tests on pull requests. Just commits. I checked your fork and they are indeed passing.

Could you add some tests to cover the case with a space and with an app name that's not all lowercase please.

Comment (Oxalin) [reply src/AutoLaunchLinux.coffee]: @4ver Latest commit should take care of the requested testcases. And we make sure to remove any files we may have created for the tests' purpose.

**Pull request merged**